met3r/met3r.py

```diff
@@ -181,36 +320,35 @@ def forward(
             feat_dissim_maps (bool[Tensor, "b h w"], optional): Feature dissimilarity score map
             proj_feats (bool[Tensor, "b h w c"], optional): Projected and rendered features
         """
-        
+
+        images = torch.zeros(1, 2, *train_rgb.shape).to(train_rgb.device)
+        images[0, 0] = ood_rgb
+        images[0, 1] = train_rgb
         *_, h, w = images.shape
-        
+
+        if K is not None or train_pose is not None or train_depth is not None:
+            assert K is not None and train_pose is not None and train_depth is not None, \
+                'K, pose, and depth_map must be provided together'
+
         # Set rasterization settings on the fly based on input resolution
         if self.img_size is None:
             raster_settings = PointsRasterizationSettings(
-                    image_size=(h, w), 
-                    radius = 0.01,
-                    points_per_pixel = 10,
-                    bin_size=0
+                image_size=(h, w),
+                radius=0.01,
+                points_per_pixel=10,
+                bin_size=0
                 )
             self.rasterizer = PointsRasterizer(cameras=None, raster_settings=raster_settings)
 
-        # K=2 since we only compare an image pairs
-        # NOTE: Apply DUST3R to get point maps and confidence scores
-        view1 = {"img": images[:, 0, ...], "instance": [""]}
-        view2 = {"img": images[:, 1, ...], "instance": [""]}
-        pred1, pred2 = self.dust3r(view1, view2)
-
-        ptmps = torch.stack([pred1["pts3d"], pred2["pts3d_in_other_view"]], dim=1).detach()
-        conf = torch.stack([pred1["conf"], pred2["conf"]], dim=1).detach()
+        if train_depth is None:
+            canon, ptmps = self._compute_canonical_point_map(images, return_ptmps=True)
+        else:
 
-        # NOTE: Get canonical point map using the confidences
-        confs11 = conf.unsqueeze(-1) - 0.999
-        canon = (confs11 * ptmps).sum(1) / confs11.sum(1)
-        
+            canon, ptmps = self._canonical_point_map_from_depth(train_depth, ood_depth, K, train_pose, ood_pose)
```

Comment:
This canonical point map is used only for focal length estimation. No need to compute it in your case.